src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1726,7 +1726,7 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         $messageToSendersMapping = [];
         $messagesToSendAndHandle = [];
         foreach ($config['routing'] as $message => $messageConfiguration) {
-            if ('*' !== $message && !class_exists($message) && !interface_exists($message, false)) {
+            if ('*' !== substr($message, -1) && !class_exists($message) && !interface_exists($message, false)) {
                 throw new LogicException(sprintf('Invalid Messenger routing configuration: class or interface "%s" not found.', $message));
```

Comment:
Might be useful to update the exception message to mention the `*` notation in case they fail to achieve something like that wtihout the *.